packages/core/admin/admin/src/content-manager/pages/EditView/components/Header.tsx

```diff
@@ -52,5 +73,177 @@ const Header = ({
+ */
+const HeaderActions = () => {
+  const [
+    {
+      query: { status = 'draft' },
+    },
+  ] = useQueryParams<{ status: 'draft' | 'published' }>();
+  const { model, id, document, meta, collectionType } = useDoc();
+  const { plugins } = useStrapiApp();
+
+  const props = {
+    activeTab: status,
+    model,
+    id,
+    document,
+    meta,
+    collectionType,
+  } satisfies DocumentActionProps;
+
+  return (
+    <Flex gap={2}>
+      <DescriptionComponentRenderer
+        props={props}
+        descriptions={(
+          plugins['content-manager'].apis as ContentManagerPlugin['config']['apis']
+        ).getDocumentActions()}
+      >
+        {(actions) => {
+          const headerActions = actions.filter((act) => act.position === 'header');
+
+          return <DocumentActionsMenu actions={headerActions} />;
+        }}
+      </DescriptionComponentRenderer>
+    </Flex>
+  );
+};
+
+/* -------------------------------------------------------------------------------------------------
+ * DocumentActionComponents
+ * -----------------------------------------------------------------------------------------------*/
+
+const ConfigureTheViewAction: DocumentActionComponent = ({ collectionType, model }) => {
+  const navigate = useNavigate();
+  const { formatMessage } = useIntl();
+
+  return {
+    label: formatMessage({
+      id: 'app.links.configure-view',
+      defaultMessage: 'Configure the view',
+    }),
+    icon: <StyledCog />,
+    onClick: () => {
+      navigate(`../${collectionType}/${model}/configurations/edit`);
+    },
+    position: 'header',
+  };
+};
+
+ConfigureTheViewAction.type = 'configure-the-view';
+
+/**
+ * Because the icon system is completely broken, we have to do
+ * this to remove the fill from the cog.
+ */
+const StyledCog = styled(Cog)`
+  path {
+    fill: none;
+    stroke: currentColor;
+  }
+`;
+
+const EditTheModelAction: DocumentActionComponent = ({ model }) => {
+  const navigate = useNavigate();
+  const { formatMessage } = useIntl();
+
+  return {
+    label: formatMessage({
+      id: 'content-manager.link-to-ctb',
+      defaultMessage: 'Edit the model',
+    }),
+    icon: <StyledPencil />,
+    onClick: () => {
+      navigate(`/plugins/content-type-builder/content-types/${model}`);
+    },
+    position: 'header',
+  };
+};
+
+EditTheModelAction.type = 'edit-the-model';
+
+/**
+ * Because the icon system is completely broken, we have to do
+ * this to remove the fill from the cog.
+ */
+const StyledPencil = styled(Pencil)`
+  path {
+    fill: none;
+    stroke: currentColor;
+  }
+`;
+
+const DeleteAction: DocumentActionComponent = ({ id, model, collectionType }) => {
+  const navigate = useNavigate();
+  const { formatMessage } = useIntl();
+  const canDelete = useDocumentRBAC('DeleteAction', (state) => state.canDelete);
+  const { delete: deleteAction } = useDocumentActions();
+
+  return {
+    disabled: !canDelete || (!id && collectionType !== SINGLE_TYPES),
```

Comment:
The disabled attribute seems to have no effect here. I was trying to adjust it
as the action is still available for single types that don't yet exist and I
think that is because of this criteria?

Reply:
smart catch!